llvm-external-projects/iree-dialects/lib/Dialect/LinalgTransform/IR/StructuredTransformOpsExt.cpp

```diff
@@ -654,17 +669,23 @@ DiagnosedSilenceableFailure transform_ext::CanonicalizedSequenceOp::apply(
   };
 
   LLVM_DEBUG(DBGS() << "begin canonicalizing sequence\n");
-  if (failed(checkedListenerTransform(performCSE)))
+  if (failed(checkedListenerTransform(performCSE))) {
+    this->emitOpError("Failed to performCSE beform transform sequence");
     return DiagnosedSilenceableFailure::definiteFailure();
-  if (failed(checkedListenerTransform(performCanonicalization)))
+  }
+  if (failed(checkedListenerTransform(performCanonicalization))) {
+    this->emitOpError(
+        "Failed to performCanonicalization beform transform sequence");
     return DiagnosedSilenceableFailure::definiteFailure();
+  }
 
   // Apply the sequenced ops one by one.
   for (Operation &transform : getBodyBlock()->without_terminator()) {
     auto transformOp = cast<transform::TransformOpInterface>(transform);
     DiagnosedSilenceableFailure result = state.applyTransform(transformOp);
     if (result.isDefiniteFailure()) {
       LLVM_DEBUG(DBGS() << "failed: " << transform << "\n");
+      transform.emitOpError("definitely failed");
```

Comment:
If the result is a definite failure, an error must have been already emitted elsewhere, this message will be redundant.